### PR TITLE
Openafs 1.9 init and 1.6 deprecation

### PR DESCRIPTION
**Pull request by spacefrogg**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [ x Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by spacefrogg** [comment]
@Mic92

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -17024,8 +17024,10 @@ julia_15 = callPackage ../development/compilers/julia/1.5.nix {
 
   openbgpd = callPackage ../servers/openbgpd { };
 
-  openafs = callPackage ../servers/openafs/1.6 { tsmbac = null; ncurses = null; };
   openafs_1_8 = callPackage ../servers/openafs/1.8 { tsmbac = null; ncurses = null; };
+  openafs_1_9 = callPackage ../servers/openafs/1.9 { tsmbac = null; ncurses = null; };
+  # Current stable release; don't backport release updates!
```
Comment (Mic92) [inline pkgs/top-level/all-packages.nix:17029]: Do you mean release updates for 1.8?

File: pkgs/servers/openafs/1.9/default.nix [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+  pname = "openafs";
+  inherit version srcs;
+
+  depsBuildBuild = [ buildPackages.stdenv.cc ];
+  nativeBuildInputs = [ autoconf automake flex libxslt libtool_2 perl
+    which yacc ] ++ optionals withDevdoc [ doxygen dblatex ];
+
+  buildInputs = [ kerberos ncurses ];
+
+  patches = [ ./bosserver.patch ./cross-build.patch ] ++ optional (tsmbac != null) ./tsmbac.patch;
```
Comment (Mic92) [inline pkgs/servers/openafs/1.9/default.nix:23]: Any plans for sending the cross build patch upstream?

File: pkgs/servers/openafs/1.9/module.nix [hunk 3]
```diff
@@ -11,33 +13,20 @@ in stdenv.mkDerivation {
...
-      sha256 = "0cdd76s1h1bhxj0hl7r6mcha1jcy5vhlvc5dc8m2i83a6281yjsa";
-    })
-  ];
-  nativeBuildInputs = [ autoconf automake flex perl bison which ] ++ kernel.moduleBuildDependencies;
+  nativeBuildInputs = [ autoconf automake flex libtool_2 perl which yacc ]
+    ++ kernel.moduleBuildDependencies;
+
+  buildInputs = [ kerberos ];
+
+  patches = [];
```
Comment (Mic92) [inline pkgs/servers/openafs/1.9/module.nix:21]: Do you want to leave this in place for easier overriding?

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (spacefrogg) [reply pkgs/top-level/all-packages.nix:17029]: Minor release updates, like from 1.6 to 1.8.

File: pkgs/servers/openafs/1.9/default.nix (hunk 2 above)
Comment (spacefrogg) [reply pkgs/servers/openafs/1.9/default.nix:23]: Not in the near future. I started and it turned out that implementing the general case is more complicated but required from upstream.

File: pkgs/servers/openafs/1.9/module.nix (hunk 3 above)
Comment (spacefrogg) [reply pkgs/servers/openafs/1.9/module.nix:21]: Yes.

**Pull request merged**